.github/workflows/service-build.yml

```diff
@@ -0,0 +1,92 @@
+name: build on maven
+
+on:
+  watch:
+    types: [started]
+  pull_request:
+    types: [opened, reopened, edited, synchronize, ready_for_review]
+  push:
+    branches:
+      - main
+      - master
+      - 1.1.x
+      - '**'
+
+  workflow_dispatch:
+
+jobs:
+  verify:
+    name: verify with maven
+    runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: "-Xmx4096m -Xms2048m -XX:MaxMetaspaceSize=4096m -Xss8m"
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up JDK 11 for x64
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        architecture: x64
+        java-version: |
+          11
+
+    - uses: s4u/maven-settings-action@v2.8.0
+      with:
+        sonatypeSnapshots: true
+
+    - name: Build the serivce verify phase
+      run: mvn -B -V clean verify -Prun-its
+
+  build:
+    name: Build and push
+    runs-on: ubuntu-latest
+    needs: verify
+    if: ${{ github.event_name == 'push' }}
+    env:
+      MAVEN_OPTS: "-Xmx4096m -Xms2048m -XX:MaxMetaspaceSize=4096m -Xss8m"
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up JDK 11 for x64
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        architecture: x64
+        java-version: |
+          11
+
+    - name: Checkout tools repo
+      uses: actions/checkout@v4
+      with:
+        repository: shokakucarrier/commonjava-images
```

Comment:
Better Commonjava repo?

Reply:
Yeah, agree with Ying.

Reply:
we need to adjust dockerfile for that first